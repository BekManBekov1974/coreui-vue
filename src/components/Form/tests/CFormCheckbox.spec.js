import { mount } from '@vue/test-utils'
import Component from '../CFormCheckbox'

const ComponentName = 'CFormCheckbox'
const wrapper = mount(Component, {
  propsData: {
    label: 'label',
    id: 'some_id',
  }
})
const customWrapper = mount(Component, {
  propsData: {
    label: 'label',
    id: 'some_id',
    wasValidated: true,
    checked: 'yes',
    trueValue: 'yes',
    falseValue: 'no',
    value: 'value',
    validFeedback: 'input is valid',
    invalidFeedback: 'input is invalid',
    tooltipFeedback: true,
    description: 'permission',
    isValid: true,
    addInputClasses: 'additional-input-class',
    addLabelClasses: 'additional-label-class',
    custom: true,
    inline: true
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly basic checkbox', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
  it('changes state correctly', () => {
    customWrapper.setProps({ checked: 'no' })
    expect(customWrapper.vm.state).toBe(false)
  })
  it('emmits correct values on check', () => {
    wrapper.find('input').setChecked()
    wrapper.find('input').setChecked(false)
    const emittedValues = wrapper.emitted()['update:checked'].map(event => {
      return event[0]
    })
    expect(emittedValues).toEqual([true, false])
  })
  it('emmits correct values on check when trueValue and falseValue are set', () => {
    customWrapper.find('input').setChecked()
    customWrapper.find('input').setChecked(false)
    const emittedValues = customWrapper.emitted()['update:checked'].map(event => {
      return event[0]
    })
    expect(emittedValues).toEqual(['yes', 'no'])
  })
})
