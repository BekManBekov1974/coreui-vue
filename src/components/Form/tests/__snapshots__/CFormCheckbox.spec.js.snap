// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFormCheckbox renders correctly 1`] = `
<div
  class="c-custom-control c-custom-control-inline c-custom-checkbox c-was-validated"
  role="group"
>
   
  <!---->
   
  <input
    class="c-custom-control-input c-is-valid additional-input-class"
    id="some_id"
    type="checkbox"
    value="value"
  />
   
  <label
    class="additional-label-class c-custom-control-label"
    for="some_id"
  >
    
        label
      
  </label>
   
  <!---->
   
  <div
    class="c-valid-tooltip"
  >
    input is valid
  </div>
   
  <div
    class="c-invalid-tooltip"
  >
    input is invalid
  </div>
   
  <small
    class="c-form-text c-text-muted c-w-100"
  >
    permission
  </small>
</div>
`;

exports[`CFormCheckbox renders correctly basic checkbox 1`] = `
<div
  class="c-form-check"
  role="group"
>
   
  <!---->
   
  <input
    class="c-form-check-input"
    id="some_id"
    type="checkbox"
    value=""
  />
   
  <label
    class="c-form-check-label"
    for="some_id"
  >
    
        label
      
  </label>
   
  <!---->
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;
